### PR TITLE
Add filtering tests for collectd/mongodb

### PR DESCRIPTION
**Pull request by keitwb**

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -21,6 +21,8 @@ def pytest_collection_modifyitems(items):
                 item.add_marker("integration")
 
 
+pytest.register_assert_rewrite("tests.helpers.verify")
```
Comment (jaysfx) [inline tests/conftest.py]: Duplicated

File: tests/helpers/metadata.py [hunk 2]
```diff
@@ -1,19 +1,18 @@
...
     """Metadata information like metrics and dimensions for a monitor"""
 
-    def __init__(self, monitor_type, included_metrics, nonincluded_metrics, metric_by_group, dims):
+    def __init__(self, monitor_type, included_metrics, nonincluded_metrics, metrics_by_group, dims):
         self.monitor_type = monitor_type
         self.included_metrics = included_metrics
-        self.nonincluded_mertics = nonincluded_metrics
-        self.all_metrics = self.included_metrics | self.nonincluded_mertics
-        self.metrics_by_group = metric_by_group
+        self.nonincluded_metrics = nonincluded_metrics
```
Comment (jaysfx) [inline tests/helpers/metadata.py]: I <3 mertics

File: tests/monitors/mongodb/mongodb_test.py [hunk 3]
```diff
@@ -52,9 +60,9 @@ def test_mongo_enhanced_metrics():
...
         with Agent.run(config) as agent:
-            assert wait_for(
-                p(has_datapoint, agent.fake_services, metric_name="gauge.collection.size"), 60
-            ), "Did not get datapoint from SendCollectionMetrics config"
-            assert wait_for(
-                p(has_datapoint, agent.fake_services, metric_name="counter.collection.commandsTime"), 60
-            ), "Did not get datapoint from SendCollectionTopMetrics config"
+            verify(
+                agent,
+                METADATA.metrics_by_group["collection"]
```
Comment (jaysfx) [inline tests/monitors/mongodb/mongodb_test.py]: Does `EXPECTED_DEFAULTS` not get received even with all enhance flags on? Or is there another reason you're ORing groups?

File: tests/conftest.py (hunk 1 above)
Comment (keitwb) [reply tests/conftest.py]: Fixed in the other PR for expvar.

File: tests/monitors/mongodb/mongodb_test.py (hunk 3 above)
Comment (keitwb) [reply tests/monitors/mongodb/mongodb_test.py]: `|` means to union all the groups together right?  I mean to test for everything coming through.

Comment (jaysfx) [reply tests/monitors/mongodb/mongodb_test.py]: What I really meant to say was all metrics minus the excluded ones:

```python
METADATA.all_metrics - {
    # These metrics only occur on MMAP storage engines.
    "gauge.backgroundFlushing.average_ms",
    "gauge.backgroundFlushing.last_ms",
    "counter.backgroundFlushing.flushes",
    # This one seems to be missing on newer mongo versions
    "gauge.extra_info.heap_usage_bytes",
}
```

**Pull request merged**